Feature/final sign

julia0926 opened the pull request:
# What is this PR? 🔍
- **이슈** : 
- **피그마** : 
- **기타 관련 문서** :
# To Reviewers 🙏

로그인, 회원가입, 프로필 부분 여태까지 작업했던 부분 한번 머지하려구 합니다!!
중간중간 SignUp 브랜치에서 PR올리고 작업했어서, 엄청나게 새로운건 없을 것 같아요 !!
API 연동이 많이 남아서, 추가 작업하겠습니당 ㅠㅠ

julia0926 committed:
:sparkles: [#8] SignIn 화면 세팅

julia0926 committed:
:art: [#8] EmailLogin 화면 구성

julia0926 committed:
:sparkles: [#8] SignIn Reactor 구현중

수정 필요

JeongMinHo committed:
:feature: [#8] signup reactor 구현

julia0926 committed:
:sparkles: [#8] textField 편집시 keyboard constraint 조정

julia0926 committed:
: art: [#8] 이메일로 회원가입 화면 및 폴더 구조 정리

julia0926 committed:
Merge branch 'develop' into feature/signIn

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.pbxproj

JeongMinHo committed:
:bug: conflict 해결

JeongMinHo committed:
:bug: conflict 해결

julia0926 committed:
:art: [#35] Email SignUp 이용약관 ScrollView 구현

julia0926 committed:
:art: [#37] 이메일 인증 화면 구성 및 이용약관 네이밍 수정

julia0926 committed:
:brick: 폴더 및 파일 네이밍 변경

julia0926 committed:
:sparkles: [#40] 약관에 모두 동의 구현

julia0926 committed:
:sparkles: [#35] 네비게이션 바 수정 및 화면 전환

julia0926 committed:
:sparkles: [#35] 화면전환 로직 수정

julia0926 committed:
:art: [#35] 프로필 사진 업로드 화면 구현

julia0926 committed:
:art: [#43] 프로필 키워드 기본 구성

julia0926 committed:
:sparkles: [#43] 프로필 키워드 추가 구현

julia0926 committed:
:sparkles: [#43] 프로필 키워드 삭제 구현

julia0926 committed:
:sparkles: [#43] 편집시 키보드 대응하여 constraint 조정

julia0926 committed:
:sparkles: [#43] 키워드 삭제시 Toast Message 구현

julia0926 committed:
:sparkles: [#43] 프로필 키워드 카운팅 / stackView 높이 고정

julia0926 committed:
🤝 [#43] 폴더구조 변경

julia0926 committed:
:bug: [#43] 키워드 삭제시 animation 처리

의견받아 사용여부 확인필요

julia0926 committed:
:bug: [#43] 코드리뷰 반영

julia0926 committed:
:sparkles: [#50 프로필 사진 circle 자르고 업로드

오픈소스 사용

julia0926 committed:
Merge pull request #45 from mash-up-kr/feature/profileKeyword

Feature/profile keyword

julia0926 committed:
Merge branch 'develop' into feature/signIn

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.pbxproj
#	DoriDori_iOS/DoriDori_iOS/AppStart/AppCoordinator.swift

julia0926 committed:
:bug: merge complict 수정

julia0926 committed:
✨ [#52] 회원가입 이메일 인증 전송

인증번호 체크 안됨

julia0926 committed:
✨ [#52] 소셜로그인 삭제

julia0926 committed:
✨ [#52] 회원가입 이메일 인증번호 확인

julia0926 committed:
✨[#50] 사진 업로드 기능 구현

julia0926 committed:
Merge branch 'feature/uploadProfile' of https://github.com/mash-up-kr/DoriDori_iOS into feature/uploadProfile

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.pbxproj
#	DoriDori_iOS/DoriDori_iOS/Info.plist
#	DoriDori_iOS/DoriDori_iOS/Scene/.DS_Store
#	DoriDori_iOS/DoriDori_iOS/Scene/Profile/Profile.storyboard
#	DoriDori_iOS/DoriDori_iOS/Scene/Profile/Profile/ProfileUploadViewController.swift

julia0926 committed:
✨ [#52] UnderLineTextField 사용

julia0926 committed:
✨ [#52] SignUp ViewModel 추가

julia0926 committed:
✨ [#52] 인증번호 인증 에러처리

julia0926 committed:
:bug: [#52] 확인버튼 색상 오류 수정

julia0926 committed:
✨ [#52] 인증번호 재전송 추가

julia0926 committed:
:art: Button title color 수정

julia0926 committed:
:bug: [#52] 코드 리뷰 반영

julia0926 committed:
Merge pull request #66 from mash-up-kr/feature/signUpAPI

Rename branch signIn -> signUp

julia0926 committed:
:art:[#67] 로그인 UI 및 구조 수정

julia0926 committed:
✨ [#67] 로그인 API 연결

julia0926 committed:
✨[#67] 비밀번호 입력화면 기본 구성

julia0926 committed:
✨[#67] UnderLineTextField 비밀번호 확인 케이스 추가

julia0926 committed:
:sparkles: [#67] 비밀번호 확인 valid check

julia0926 committed:
Merge branch 'feature/password' into feature/signIn

julia0926 committed:
✨[#67] 비밀번호 에러 처리

확실하지 않고 약간 수정 필요할듯..

julia0926 committed:
:art: [#72] 이용약관 화면 수정

폰트 추가 / ScrollView -> TextView 변경

julia0926 committed:
✨[#72] 이용약관 API 연결

julia0926 committed:
✨ [#72] 이용약관 조건에 따른 활성화 구현

julia0926 committed:
✨ [#50] 사진 권한 추가

julia0926 committed:
✨[#79] 닉네임 설정 화면 구현

julia0926 committed:
:art: [#79] 프로필 소개 화면 추가 및 폰트 수정

julia0926 committed:
✨ [#79] underLineTextField 닉네임, 프로필 대응하여 수정

julia0926 committed:
:truck: [#79] 프로필 폴더 구조 변경 및 파일 추가

julia0926 committed:
✨ [#79] 프로필 소개 구현

julia0926 committed:
:recycle: [#79] 프로필 키워드 추가 리팩토링

julia0926 committed:
:recycle: [#79] 프로필 키워드 카운팅 수정 및 버튼 활성화

julia0926 committed:
Merge branch 'feature/signIn' into feature/finalSign

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.pbxproj
#	DoriDori_iOS/DoriDori_iOS/Scene/Common/TextField/UnderLineTextField/UnderLineTextField.swift
#	DoriDori_iOS/DoriDori_iOS/Scene/Common/TextField/UnderLineTextField/UnderLineTextFieldViewModel.swift

julia0926 committed:
Merge branch 'feature/terms' into feature/finalSign

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.pbxproj
#	DoriDori_iOS/DoriDori_iOS/Scene/Sign/SignUp/TermsOfService/TermsOfServiceViewContoller.swift

julia0926 committed:
Merge branch 'feature/uploadProfile' into feature/finalSign

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS/Common/Network/DevDatas.swift
#	DoriDori_iOS/DoriDori_iOS/Scene/Profile/ProfileUpload/ProfileUploadViewController.swift

julia0926 committed:
✨ [#67] 이메일 회원가입 서버 에러메세지 연결

SongSeoYoung commented on DoriDori_iOS/DoriDori_iOS/Common/Extensions/UIViewController+Extenstion.swift:
```diff
@@ -0,0 +1,21 @@
+import UIKit
+
+extension UIViewController {
+    
+    func configureSignUpNavigationBar() {
+        navigationItem.title = "입장하기"
+        navigationController?.navigationBar.topItem?.title = ""
+    }
+    
+    open override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        self.view.endEditing(true)
+    }
```
이거는 각자 쓰는 파일에 있었음 좋겠어요.. 
입장하기의 title이 있는 viewcontroller의 extension에 넣어주세요!!
전체 객체의 extension은 정말 프로젝트 전반에서 쓰일것들만 추가 부탁드립니당~

julia0926 replied:
넹 !! 혹시 밑에 touchesBegan 이건 제생각엔 자주 쓰일 것 같은데,, 그래도 확실하게  텍스트필드 있는 곳만 있는 쪽으로 수정할까요 ?!

SongSeoYoung commented on DoriDori_iOS/DoriDori_iOS/Common/Network/DevDatas.swift:
```diff
@@ -0,0 +1,19 @@
+//  Created by Seori on 2022/08/12.
+//
+
+import Foundation
+import Alamofire
+
+//typealias HTTPHeaders = Alamofire.HTTPHeaders
+
+let devUserID: String = "62d7f4776ad96c51d4330ea2"
+let dummyAccessToken: String = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJiYW5nd2lkYWUiLCJleHAiOjE3NTQyMzE0MTQsInVzZXJJZCI6IjYyZDdmNDc3NmFkOTZjNTFkNDMzMGVhMiJ9.qYld9Je775prztT4oGWZ-4FDYg27TVJ24h1mQZG0fiE"
+extension Requestable {
+    var headers: HTTPHeaders? {
+        ["Authorization": "Bearer \(dummyAccessToken)"]
+    }
+}
```
이게 어떻게 깃에 올라왔찌..?ㅜ

SongSeoYoung replied:
깃 이그노어 한 번 받아야할 것 같아요. !!
보니까 옛날 브랜치에서 계속 끌어다써서 그런거같은데ㅠ 디벨롭 한 번 받아와주세용~

julia0926 replied:
네에 !!

SongSeoYoung reviewed (approved):
수고하셨어요!!!__!!!! 👍

julia0926 committed:
Merge branch 'develop' into feature/finalSign

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.pbxproj
#	DoriDori_iOS/DoriDori_iOS/Common/Network/EmptyModel.swift
#	DoriDori_iOS/DoriDori_iOS/Common/Network/ErrorModel.swift
#	DoriDori_iOS/DoriDori_iOS/Common/Network/Network.swift
#	DoriDori_iOS/DoriDori_iOS/Info.plist

SongSeoYoung committed:
Merge branch 'develop' into feature/finalSign

julia0926 committed:
Merge branch 'develop' into feature/finalSign

julia0926 committed:
:bug: 코드리뷰 반영

julia0926 committed:
✨ [#82] 회원가입 API 구현 (테스트X)

- 회원가입 API를 요청하기 위해 공통으로 쓰일 SignUpViewModel을 구현
- 마지막 PasswordVC에서 요청(으로 일단 해놨으나 마지막 도리도리 시작때 가입)

julia0926 committed:
:bug: Token 옵셔널 제거

julia0926 committed:
✨[#89] 닉네임 API 연결

julia0926 committed:
Merge branch 'develop' into feature/signAPI

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS/Common/UserDefault/UserDefault+Value.swift

julia0926 committed:
:zap: 회원가입 API 리턴 타입 변경

julia0926 committed:
✨[#82] 프로필 API 연동

julia0926 committed:
✨[82] 회원가입 API 연동

julia0926 committed:
Merge branch 'develop' into feature/signAPI

# Conflicts:
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.pbxproj
#	DoriDori_iOS/DoriDori_iOS.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
#	DoriDori_iOS/DoriDori_iOS/AppStart/AppCoordinator.swift
#	DoriDori_iOS/DoriDori_iOS/AppStart/SceneDelegate.swift
#	DoriDori_iOS/DoriDori_iOS/Info.plist
#	DoriDori_iOS/DoriDori_iOS/Scene/WebView/BaseWebViewController.swift

julia0926 committed:
✨[#82] 키워드 삭제, 이메일 재전송시 토스트 추가

julia0926 committed:
✨[#82] SignIn TextField 수정

julia0926 committed:
:art: 회원가입, 로그인 폰트 수정

julia0926 committed:
:bug: [#82] signUp TextField 수정

julia0926 committed:
✨[#82] 편집시 키보드 설징 및 비밀번호 글자수 세기 추가

julia0926 committed:
✨ [#82] 이메일 인증시 인디케이터 추가

julia0926 committed:
✨[#82] 프로필 기본 설정 API 추가

julia0926 committed:
✨[#82] 이메일 회원가입 인증번호 타이머 구현

재전송시 타이머 여러번 발생하는 오류있음

julia0926 committed:
:bug: 화면전환시 이상하게 바뀌는 이슈 수정
